lib/services/android-resources-migration-service.ts

```diff
@@ -0,0 +1,57 @@
+import * as path from "path";
+import * as shell from "shelljs";
+import * as constants from "../constants";
+
+export class AndroidResourcesMigrationService implements IAndroidResourcesMigrationService {
+	private static ANDROID_DIR = "Android";
+	private static ANDROID_DIR_TEMP = "Android-Updated";
+	private static ANDROID_DIR_OLD = "Android-Pre-v4";
+
+	constructor(private $fs: IFileSystem,
+		private $logger: ILogger,
+		private $devicePlatformsConstants: Mobile.IDevicePlatformsConstants) { }
+
+	public canMigrate(platformString: string): boolean {
+		return platformString.toLowerCase() === this.$devicePlatformsConstants.Android.toLowerCase();
+	}
+
+	public hasMigrated(appResourcesDir: string): boolean {
+		return this.$fs.exists(path.join(appResourcesDir, AndroidResourcesMigrationService.ANDROID_DIR, constants.SRC_DIR, constants.MAIN_DIR));
+	}
+
+	public async migrate(appResourcesDir: string): Promise<void> {
+		const originalAppResources = path.join(appResourcesDir, AndroidResourcesMigrationService.ANDROID_DIR);
+		const appResourcesDestination = path.join(appResourcesDir, AndroidResourcesMigrationService.ANDROID_DIR_TEMP);
+		const appMainSourceSet = path.join(appResourcesDestination, constants.SRC_DIR, constants.MAIN_DIR);
+		const appResourcesMainSourceSetResourcesDestination = path.join(appMainSourceSet, constants.RESOURCES_DIR);
+
+		this.$fs.ensureDirectoryExists(appResourcesDestination);
+		this.$fs.ensureDirectoryExists(appMainSourceSet);
+		// create /java, /res and /assets in the App_Resources/Android/src/main directory
+		this.$fs.ensureDirectoryExists(appResourcesMainSourceSetResourcesDestination);
+		this.$fs.ensureDirectoryExists(path.join(appMainSourceSet, "java"));
+		this.$fs.ensureDirectoryExists(path.join(appMainSourceSet, constants.ASSETS_DIR));
+
+		const isDirectory = (source: string) => this.$fs.getLsStats(source).isDirectory();
+		const getDirectories = (source: string) =>
+			this.$fs.readDirectory(source).map(name => path.join(source, name)).filter(isDirectory);
+
+		this.$fs.copyFile(path.join(originalAppResources, "app.gradle"), path.join(appResourcesDestination, "app.gradle"));
```

Comment:
Please take in consideration other files that might be inside the root of the App_Resources directory like "settings.gradle" and "settings.josn".